platform/consul/conversion.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consul
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"istio.io/pilot/model"
+)
+
+const (
+	protocolTagName = "protocol"
+	externalTagName = "external"
+)
+
+func convertTags(tags []string) model.Tags {
+	out := make(model.Tags, len(tags))
+	for _, tag := range tags {
+		vals := strings.Split(tag, "=")
+		if len(vals) > 1 {
+			out[vals[0]] = vals[1]
+		} else {
+			// TODO safe to assume all tags are of form "key=value"?
+			out[tag] = ""
+		}
+	}
+	return out
+}
+
+func convertPort(port int, name string) *model.Port {
+
+	// TODO default HTTP/TCP?
```

Comment:
According to the `model` package, you need to set a name if you have multiple ports for a single service:
```go
type Port struct {
	// Name ascribes a human readable name for the port object. When a
	// service has multiple ports, the name field is mandatory
	Name string `json:"name,omitempty"
...
```